fdbclient/GlobalConfig.actor.h

```diff
@@ -77,23 +80,39 @@ class GlobalConfig : NonCopyable {
 	// database.
 	template <class T>
 	static void create(Database& cx, Reference<AsyncVar<T> const> db, const ClientDBInfo* dbInfo) {
-		if (g_network->global(INetwork::enGlobalConfig) == nullptr) {
+		auto config_map =
+		    reinterpret_cast<ConfigMap*>(g_network->global(INetwork::enGlobalConfig));
+		if (config_map == nullptr) {
+			auto m = new ConfigMap();
+			g_network->setGlobal(INetwork::enGlobalConfig, m);
+			config_map = m;
+		}
+
+		auto it = config_map->find(cx.dbId());
+		if (it == config_map->end()) {
 			auto config = new GlobalConfig{ cx };
 			g_network->setGlobal(INetwork::enGlobalConfig, config);
 			config->_updater = updater(config, dbInfo);
 			// Bind changes in `db` to the `dbInfoChanged` AsyncTrigger.
 			// TODO: Change AsyncTrigger to a Reference
 			forward(db, std::addressof(config->dbInfoChanged));
 		} else {
-			GlobalConfig* config = reinterpret_cast<GlobalConfig*>(g_network->global(INetwork::enGlobalConfig));
+			GlobalConfig* config = it->second;
 			config->cx = cx;
+			config->_updater = updater(config, dbInfo);
 		}
 	}
 
+	static GlobalConfig& globalConfig() {
+		ASSERT(false);
+		return *reinterpret_cast<GlobalConfig*>(0);
+	}
```

Comment:
Remove this? Anywhere this is called should be replaced with the other two with parameters.